Library/Homebrew/os/mac/sdk.rb

```diff
@@ -91,6 +91,14 @@ def sdk_paths
               paths[OS::Mac::Version.new(version)] = sdk_path if version.present?
             end
 
+            # Use unversioned SDK path on Big Sur to avoid issues such as:
+            # https://github.com/Homebrew/homebrew-core/issues/67075
+            if OS::Mac.version >= :big_sur
+              sdk_path = File.join(sdk_prefix, "MacOSX.sdk")
+              version = OS::Mac.full_version
+              paths[version] = sdk_path if File.directory?(sdk_path)
```

Comment:
I think this should be `OS::Mac.sdk_version` rather than `full_version` or there will be problems should there be a 11.1.1.

But I actually think the best solution here is _not_  use the runtime MacOS version at all. That's open to edge cases, particularly with outdated CLT installs. This strongly assumes the user is running a CLT/Xcode version released around the same time as their macOS version. This can very easily not be the case. In the case of major macOS versions, we absolutely do not want old SDKs being seen as compatible with the latest OS - we even have `brew doctor` checks that go through the SDK versions detected here.

I strongly suggest either:

* Reading the symlink and detecting the SDK version from the real path, or
* Reading the plist in the SDK and extracting the version from that

This should also alleviate some of the concerns in the earlier review.

Reply:
>I think this should be `OS::Mac.sdk_version` rather than `full_version` or there will be problems should there be a 11.1.1.

Yes, I agree.

>But I actually think the best solution here is to not use the runtime MacOS version at all. That's open to edge cases, particularly with outdated CLT installs.

Aren't outdated CLT installs already picked up by `brew doctor`? I should also mention that the Big Sur 11.1 update actually wiped out my CLT installation. If Apple keeps doing that, then this may not be too significant a problem. (Big "if", I know.)

>we absolutely do not want old SDKs being seen as compatible with the latest OS

It seems we already started being OK with that in the case of Big Sur though. We also have other code that allows for minor-version mismatches in the SDK. I agree that we do not want, say, `MacOSX10.14.sdk` being picked up on Big Sur, but I'd need to think more carefully about circumstances when this might happen but isn't caught otherwise.

>we even have `brew doctor` checks that go through the SDK versions detected here.

I was concerned about this too, which is why I had a look before this PR was merged at `brew doctor` to see if this change would affect its CLT version detection information. I don't believe it does, but I'd love a second opinion on this.

>Reading the symlink and detecting the SDK version from the real path

This only works for CLT. For Xcode, it is `MacOSX11.1.sdk` that is the symlink to `MacOSX.sdk`, and not vice-versa. I guess we could make it work by having two different implementations of `sdk_paths` in `XcodeSDKLocator` and `CLTSDKLocator`, but I suspect that is not a good solution.

>Reading the plist in the SDK and extracting the version from that

I'll have a look at this one.

Reply:
It seems `/Library/Developer/CommandLineTools/SDKs/MacOSX.sdk` contains (among others) `SDKSettings.plist` and `SDKSettings.json`. It appears that the contents of the two are identical. In particular, both have an entry whose key is "Version" that we can use.

@fxcoudert, can you confirm the same for Xcode?

Reply:
@carlocab Confirmed:

```console
$ cat /Applications/Xcode.app/Contents/Developer/Platforms/MacOSX.platform/Developer/SDKs/MacOSX.sdk/SDKSettings.json | jq .Version
"11.0"
```

@carlocab Would you be able to create a follow-up PR given the feedback above? Thanks!

Reply:
Thanks, @MikeMcQuaid. On it.

Reply:
Only extra thing to remember is that the 10.15 SDK might(?) have 3 components in the version so might need to strip that off to be just 2. I know a 10.15.4 SDK was released at some point.

Reply:
> Aren't outdated CLT installs already picked up by `brew doctor`?

Only after changes are made in `brew`. For example, when the Big Sur betas came out, people were already trying it before `brew` got Big Sur support and reported cryptic errors caused by outdated CLTs. So we added a general check for SDK versions to prevent that happening in the future.

> It seems we already started being OK with that in the case of Big Sur though.

Sure, but it won't be okay once macOS 12 SDK lands.